Updated UI_OxygenAlert to use UpdateManager

plnqux opened the pull request:
### Purpose
Updated UI_OxygenAlert to use the UpdateManager to remove an instance of InvokeRepeating.

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### Notes:
While working on #2121 I found that the UI_OxygenAlert was using InvokeRepeating and saw a way to make this class use the UpdateManager instead based on the UI_HeartMonitor class.  I'm not sure what all the cons are for using InvokeRepeating but one I thought of was that the CycleImg function could have been called when the O2 alert was not on screen.  I believe using OnEnable and OnDisable to register the UpdateMe function will prevent that.

plnqux committed:
Updated UI_OxygenAlert to use UpdateManager

RedlineTriad commented on UnityProject/Assets/Scripts/UI/UI_OxygenAlert.cs:
```diff
@@ -10,12 +10,36 @@ public class UI_OxygenAlert : MonoBehaviour {
+	{
+		timeWait += Time.deltaTime;
+		// Cycle images every 1 second
+		if (timeWait > 1f)
```
The value that `timeWait` is compared to should either be a constant or a serialized value.
It's easier to understand and modify if there are fewer magic numbers.

plnqux replied:
I've created a constant now.

RedlineTriad commented on UnityProject/Assets/Scripts/UI/UI_OxygenAlert.cs:
```diff
@@ -10,12 +10,36 @@ public class UI_OxygenAlert : MonoBehaviour {
+		if (timeWait > 1f)
+		{
+			CycleImg();
+			timeWait = 0f;
```
This would technically only be accurate with infinite FPS, because when we have, say 400ms per frame it would
```
timeWait = 0;
timeWait = 0.4;
timeWait = 0.8;
timeWait = 1.2 -> 0 // Cycle image called;
```
This means that that the lower the FPS the more inaccurate the cycle time is.
Should be changed to `timeWait -= CycleTime;`

plnqux replied:
I've updated the code to subtract the new constant instead of setting timeWait to 0f.

plnqux committed:
Updated UI_OxygenAlert for PR fixes and reset cycling on disable

Added constant for cycle time
Updated to subract cycle time from timeWait when cycling the image
Changed activeImageIndex to nextImageIndex and set it to 1 initially, component was keeping the same image for two cycles when first shown
Added reset function to reset the cycle when disabling the component

RedlineTriad reviewed (approved):
I would merge this however I am currently very scared of switching branches due to some Git/Linux/Unity issues, so I can't test the code running in unity.

fomalsd reviewed (commented):
it works, but: it can be simpler if you take advantage of our `array.Wrap()` [extension method](https://github.com/unitystation/unitystation/blob/ce52efc6c589888101d389b2c50bc1cde53f317e/UnityProject/Assets/Scripts/Util/SweetExtensions.cs#L65), also making local `sprite` field redundant. 
see for yourself: https://gist.github.com/fomalsd/f662a3995805bb44cf2e1f65327f24f1

plnqux committed:
Simplified based on gist using array.Wrap
